### PR TITLE
Bug 1819254 - Don't exit fullscreen for user input prompts

### PR DESCRIPTION
**Pull request by Mugurell**
Showing the YouTube issue fixed:

https://user-images.githubusercontent.com/11428869/223170127-e7a11f1b-1729-4697-ae04-8d67f5b2fb52.mp4

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1819254

### CODE REVIEW

**Comment by Mugurell** [comment]
cc @emilio

**Comment by Mugurell** [comment]
Emilio confirmed "offline" that [the proposed list of prompts ](https://github.com/mozilla-mobile/firefox-android/blob/f8cd8f82c24ca5fc35e26b5bead244d66d740f41/android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/ext/PromptRequest.kt) looks okay.

File: android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/ext/PromptRequest.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+@PublishedApi
+internal val PROMPTS_TO_EXIT_FULLSCREEN_FOR = listOf<KClass<out PromptRequest>>(
+    Alert::class,
+    TextPrompt::class,
+    Confirm::class,
+    Popup::class,
+)
+
+/**
+ * Convenience method for executing executing code if the current [PromptRequest] is one that
```
[the remark this answers is not in the document]
Comment (Mugurell) [reply android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/ext/PromptRequest.kt]: Thank you! 
Fixed.

**Review by Amejia481 — APPROVED** [review]
LGMT!

**Comment by Mugurell** [comment]
@Mergifyio backport releases_v111

**Pull request merged**